### PR TITLE
selinux: upgrade selinux libs

### PR DESCRIPTION
**Pull request by tormath1**
this PR is based on https://github.com/kinvolk/coreos-overlay/pull/347 but it has been adapted to match flatcar git commit workflow and to clean the git history

:warning: to be merged with https://github.com/kinvolk/portage-stable/pull/172

as it's quite a big PR, we can split it into smaller ones

## Note for reviewers

- `setools` upgrade from 3 to 4 is fun (https://github.com/SELinuxProject/setools/wiki/Changes-Since-SETools-v3) (rewrite from C to Python). ~IDK yet how we can include this scripts into the OS :thinking:~ we won't ship it anymore 

## Todo

- remove script with python requirements
```
/usr/share/selinux/targeted/include/support/segenxml.py: (env)/python3 does not exist
/usr/share/selinux/mls/include/support/segenxml.py: (env)/python3 does not exist
/usr/share/selinux/mcs/include/support/segenxml.py: (env)/python3 does not exist
```
(according to selinux repository, this scripts should only be "helpers")
- `selinux-unconfined` seems to be able to move into `::portage-stable`

## Testing done

- booting the qemu instance
- kola tests (https://github.com/kinvolk/mantle/pull/177) 
- Jenkins CI: http://jenkins.infra.kinvolk.io:8080/job/os/job/manifest/2927/cldsv/

closes https://github.com/kinvolk/Flatcar/issues/305, (and certainly other SELinux related issues - need to pass through)

### CODE REVIEW

**Comment by dongsupark** [comment]
Build fails like that:
```
 * Applying mcs-sshd.patch ...
/build/amd64-usr/var/tmp/portage/sec-policy/selinux-base-2.20200818-r2/temp/environment: line 518: /build/amd64-usr/var/tmp/portage/sec-policy/selinux-base-2.20200818-r2/files/mcs-sshd.patch: No such file or directory
/build/amd64-usr/var/tmp/portage/sec-policy/selinux-base-2.20200818-r2/temp/environment: line 521: /build/amd64-usr/var/tmp/portage/sec-policy/selinux-base-2.20200818-r2/files/mcs-sshd.patch: No such file or directory
```

**Comment by tormath1** [comment]
@dongsupark thanks, I forgot to remove this patch. Actually it has been moved from `selinux-base` to `selinux-base-policy`. Otherwise it was override.
CI is currently failing because of glibc, I need to investigate on it :thinking:

**Comment by tormath1** [comment]
Rebased with `main` to include glibc stuffs - otherwise the CI was failing :)

File: sys-apps/policycoreutils/policycoreutils-3.1-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+	if use python; then
+		python_copy_sources
+		# Our extra code is outside the regular directory, so set it to the extra
+		# directory. We really should optimize this as it is ugly, but the extra
+		# code is needed for Gentoo at the same time that policycoreutils is present
+		# (so we cannot use an additional package for now).
+		if use extra ; then
+			S="${S2}"
+			python_copy_sources
+		fi
```
Comment (dongsupark) [inline sys-apps/policycoreutils/policycoreutils-3.1-r2.ebuild:101]: Just thought.
Ideally we should contribute to Gentoo code like that, making USE flags to exclude unnecessary parts.
Of course that would be a long term project. So for now it is ok.

**Review by dongsupark — COMMENTED** [review]
Thanks. Code looks good in general. 
My local test looks ok as well.

As for cosmetic issues, can you please rearrange commits, e.g.,
squash fixup commits, move Flatcar patch commits right next to their corresponding Gentoo sync commits, etc?

File: sys-apps/policycoreutils/policycoreutils-3.1-r2.ebuild (hunk 1 above)
Comment (tormath1) [reply sys-apps/policycoreutils/policycoreutils-3.1-r2.ebuild:101]: I'm not a big fan of this `extra` USE flag - but yeah we could definitely try to provide more granular control on this kind of part.

**Comment by tormath1** [comment]
@dongsupark thanks for the review - commits squash and rearranged :)

**Review by dongsupark — APPROVED** [review]

**Pull request merged**